atom/browser/web_contents_preferences.cc

```diff
@@ -197,6 +193,8 @@ bool WebContentsPreferences::IsEnabled(const base::StringPiece& name,
 void WebContentsPreferences::Merge(const base::DictionaryValue& extend) {
   if (preference_.is_dict())
     static_cast<base::DictionaryValue*>(&preference_)->MergeDictionary(&extend);
+
+  SetDefaults();
```

Comment:
The constructor already has defaults set, so I thought calling `SetDefaults` after `Merge` will not have any effect as `MergeDictionary` won't be deleting properties?

Reply:
In this case we call `Clear` before `Merge`
https://github.com/electron/electron/blob/master/atom/browser/api/atom_api_browser_window.cc#L61

Reply:
also some property's default value can depend on other property's value, which can be modified by the merge operation.

Reply:
Thanks for clarification.